spec/ddtrace/workers_integration_spec.rb

```diff
@@ -213,6 +215,14 @@ def wait_for_flush(num = 1, period = 0.1)
       )
     end
 
+    after do
+      thread = worker.instance_variable_get(:@worker)
+      if thread
+        thread.terminate
+        thread.join
+      end
+    end
```

Comment:
I'm curious -- would `worker.join` work here? Since we're already calling `.stop`  in `before`, is there any situation where the thread would just get stuck?

Reply:
The timeout case is never forced to terminate with `.stop`, thus waits for the thread to terminate (10 seconds in the test setup).

```ruby
context 'which overruns the timeout' do
      let(:task) { proc { sleep(interval) } }
```

Reply:
Ah, I see. Upon closer look, indeed the `stop` just signals the thread to end, but if it's stuck in the sleep it will only see the signal after it wakes up. 👍